beacon-chain/rpc/eth/beacon/handlers.go

```diff
@@ -959,6 +959,54 @@ func (s *Server) publishBlock(ctx context.Context, w http.ResponseWriter, r *htt
 	httputil.HandleError(w, "Body does not represent a valid block type", http.StatusBadRequest)
 }
 
+func (s *Server) PublishBlobs(w http.ResponseWriter, r *http.Request) {
+	ctx, span := trace.StartSpan(r.Context(), "beacon.PublishBlobs")
+	defer span.End()
+	if shared.IsSyncing(r.Context(), w, s.SyncChecker, s.HeadFetcher, s.TimeFetcher, s.OptimisticModeFetcher) {
+		return
+	}
+
+	var req structs.PublishBlobsRequest
+	if err := json.NewDecoder(r.Body).Decode(&req); err != nil {
+		httputil.HandleError(w, "Could not decode JSON request body", http.StatusBadRequest)
+		return
+	}
+
+	root, err := bytesutil.DecodeHexWithLength(req.BeaconRoot, 32)
+	if err != nil {
+		httputil.HandleError(w, "Could not decode beacon root", http.StatusBadRequest)
+		return
+	}
+	
+	var beaconRoot [32]byte
+	copy(beaconRoot[:], root)
+
+	for _, blobSidecar := range req.BlobSidecars.Sidecars {
+
+		sc, err := blobSidecar.ToConsensus()
+		if err != nil {
+			http.Error(w, "Could not decode blob sidecar", http.StatusBadRequest)
+			return
+		}
+
+		readOnlySc, err := blocks.NewROBlobWithRoot(sc, beaconRoot)
```

Comment:
we have a helper function bytesutil.ToBytes32 maybe don't need lines 981/982

Reply:
Updated